engine/consensus/sealing/core.go

```diff
@@ -169,6 +183,11 @@ func (c *Core) RepopulateAssignmentCollectorTree(payloads storage.Payloads) erro
 		}
 
 		for _, result := range payload.Results {
+			// skip results referencing blocks before the root sealing segment
+			_, isOutdated := outdatedBlockIDs[result.BlockID]
+			if isOutdated {
+				continue
```

Comment:
If this result is referring to an old block, better to add a WARN log with both the result ID and the blockID

Reply:
https://github.com/onflow/flow-go/pull/1828/commits/443ed762c03f5924a8d8afeb1575d82b17ddd799